Generic File Upload

dtext opened the pull request:
Allows users to upload and share files present on the local file system. The changes in this need to be implemented in #34 and #43, so we should definitely merge this first.
Edit: merged #43 already, order doesn't really matter here.

### Commit Summary
The first commit moves a lot of code from the specific subcommands to the `storage` package. The reasoning behind this is explained in the commit message below.

The second commit just adds the few lines of code that were then needed to implement sharing of existing files.

### Relevant issues
Closes #2, closes #11

dtext committed:
Refactor uploading to avoid code duplication

All the code after taking a screenshot was moved to the storage package.
Strictly speaking, the package does more than just storage now, but all
functionality is about uploading and what happens after that
(like notifications, clipboard), so it makes much more sense to put it
there.

dtext committed:
Add the capability to instantly share generic files

dtext committed:
Fix copy paste error

denniseffing commented on src/cli/file.py:
```diff
@@ -1,7 +1,24 @@
+    file = args["<filename>"]
+
+    if hoster:
+        storage.upload_to(hoster, file)
```
What happens here if the user enters an invalid combination of hoster and file?

dtext replied:
Any checks of that kind would need to be done inside the different storage modules. We don't really have proper error handling for that yet.

denniseffing commented on src/tools/toolbox.py:
```diff
@@ -32,3 +33,7 @@ def __init__(self):
+
+
+def build_filename(media_type: str):
+    return "{}/instantshare_{}_{}.png".format(gettempdir(), media_type, strftime("%Y-%m-%d_%H-%I-%S"))
```
Calling `build_filename` with `media_type=text` results in a file named `<tempdir>/instantshare_<media_type>_<timestamp>.PNG` which only makes sense for screenshots.

dtext replied:
That's right. I will see what I can come up with.

denniseffing reviewed (changes requested):
Solid implementation, even though I dislike the new multi-purpose of our storage module (but it is better than before). I don't have a better solution though, so this has to do for now.

dtext committed:
Merge branch 'master' into ft/any-file-upload

dtext committed:
Fix the build_filename function

The build_filename function, which is used to create a name for a shared
file using a well defined scheme, was updated to support multimedia and
was moved to a more fitting package.
a more fitting module

dtext commented:
@denniseffing What do you think about this? We know the file name and extension for every file that is created by us (screenshots, recordings, etc.). For existing files being shared (which could have any file extension), we don't even need to set a filename anyway.

dtext committed:
Adjust file names for shared files.
